### ISSUE TITLE
Initializing TaskDescription from a dictionary fails on *metadata* field.

### ISSUE THREAD

**Issue by eirrgang** [radical-cybertools/radical.pilot#2797]
I can create a TaskDescription fine this way:
```python
            td = rp.TaskDescription()
            td.scheduler = scheduler.uid
            td.mode = a_string
            td.metadata = a_dict
```

but when I try this way
```python
            td = rp.TaskDescription(
                from_dict={
                    'scheduler': scheduler.uid,
                    'mode': a_string,
                    'metadata': a_dict
                }
            )
```

I get an error while processing the metadata key:
```
Traceback (most recent call last):
  File "/tmp/scalems_dev/src/scalems/workflow/__init__.py", line 639, in dispatch
    yield dispatching_session
  File "/tmp/scalems_dev/tests/test_rp_exec.py", line 91, in test_raptor_master
    td = rp.TaskDescription(
  File "/home/rp/rp-venv/lib/python3.9/site-packages/radical/pilot/task_description.py", line 621, in __init__
    super().__init__(from_dict=from_dict)
  File "/home/rp/rp-venv/lib/python3.9/site-packages/radical/utils/typeddict.py", line 137, in __init__
    self.update(from_dict)
  File "/home/rp/rp-venv/lib/python3.9/site-packages/radical/utils/typeddict.py", line 173, in update
    self.setdefault(k, t()).update(v)
AttributeError: 'NoneType' object has no attribute 'update'
```

## More debuggging info:
```
        for k, v in other.items():
            if isinstance(v, dict):
                t = self._schema.get(k) or \
                    (type(self) if self._self_default else TypedDict)
                if isinstance(t, type) and issubclass(t, TypedDict):
                    # cast to expected TypedDict type
>                   self.setdefault(k, t()).update(v)
E                   AttributeError: 'NoneType' object has no attribute 'update'

k          = 'metadata'
other      = {'metadata': {'control': 'hello'}, 'mode': 'scalems.cpi', 'scheduler': 'scalems-rp-master.7fc4f484-7af6-11ed-8e20-0242ac120003'}
self       = <TaskDescription object, schema keys: ('uid', 'name', 'mode', 'executable', 'arguments', 'code', 'function', 'args', '...hread_type', 'lfs_per_process', 'mem_per_process', 'restartable', 'scheduler', 'tags', 'metadata', 'cleanup', 'pilot')>
t          = <class 'radical.utils.typeddict.TypedDict'>
v          = {'control': 'hello'}

/home/rp/rp-venv/lib/python3.9/site-packages/radical/utils/typeddict.py:173: AttributeError
```

```
~$ pip list |grep radical
radical.gtod       1.16.0
radical.pilot      1.18.1
radical.saga       1.18.0
radical.utils      1.18.1
```

**Comment by mtitov** [comment]
@eirrgang Eric, thank you for revealing this bug, it is fixed in the `radical.utils` devel-branch

**Comment by andre-merzky** [comment]
Just to note: this was released in 1.20.0 today.
